src/VisualStudio/TestUtilities2/MockVisualStudioDiagnosticAnalyzerProviderFactory.vb

```diff
@@ -15,13 +15,15 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Diagnostics
 
         Public Property Extensions As (Paths As String(), Id As String)() = Array.Empty(Of (Paths As String(), Id As String))()
```

Comment:
Wouldn't it be better to let callers specify the content type in the tuple here, so it is possible to mock the razor extension as containing both Analyzer and Razor items? Or would multiple mock providers for the same extension with different content types work equivalently?

Reply:
It doesn't seem worth it. All we'd actually be testing is the mock itself. We could make it work by just removing the assert in the mock that the type being retrieved is the type passed into the constructor, but that at least seem slightly useful to make sure the correct mock is being used.